File: Front-end/app/common/service/report.service.js
export default function reportService($http) {
    var service = this;

    service.getReportsByEmployeeId = function (employeeId) {
        return $http({
            method: "GET",
            url: EM_REST_SERVER_URI + "/employees/" + employeeId + "/reports"
        }).then(function (response) {
            return response.data;
        });
    };

    service.getReportByEmployeeIdAndTestPeriodId = function (employeeId, testPeriodId) {
        return $http({
            method: "GET",
            url: EM_REST_SERVER_URI + "/employees/" + employeeId + "/reports/" + testPeriodId
        }).then(function (response) {
            return response.data;
        });
    };
}

File: Back-end/src/main/java/by/bsuir/em/service/ReportService.java
package by.bsuir.em.service;

import by.bsuir.em.dto.ReportDto;

import java.util.List;

public interface ReportService {
    List<ReportDto> getReportsByEmployeeId(Long employeeId);
    ReportDto getReportByEmployeeIdAndTestPeriodId(Long employeeId, Long testPeriodId);
}
